src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -697,28 +727,52 @@ class EthService(
     }
   }
 
-  def getTransactionByBlockNumberAndIndexRequest(
+  /**
+    * eth_getTransactionByBlockNumberAndIndex Returns the information about a transaction with
+    * the block number and index of which it was mined.
+    *
+    * @param req block number and index
+    * @return transaction
+    */
+  def getTransactionByBlockNumberAndIndex(
       req: GetTransactionByBlockNumberAndIndexRequest
   ): ServiceResponse[GetTransactionByBlockNumberAndIndexResponse] = Future {
-    import req._
+    getTransactionDataByBlockNumberAndIndex(req.block, req.transactionIndex)
+      .map(_.map(TransactionResponse(_)))
+      .map(GetTransactionByBlockNumberAndIndexResponse)
+  }
+
+  /**
+    * eth_getRawTransactionByBlockNumberAndIndex Returns raw transaction data of a transaction
+    * with the block number and index of which it was mined.
+    *
+    * @param req block number and ordering in which a transaction is mined within its block
+    * @return raw transaction data
+    */
+  def getRawTransactionByBlockNumberAndIndex(
+    req: GetTransactionByBlockNumberAndIndexRequest
+  ): ServiceResponse[RawTransactionResponse] = Future {
+    getTransactionDataByBlockNumberAndIndex(req.block, req.transactionIndex)
+      .map(x => x.map(_.stx))
+      .map(RawTransactionResponse)
```

Comment:
minor: `asRawTransactionResponse`?

Reply:
Types do not match. It should be possible to fix that but it would require an invasive change in existing endpoint implementation. I would prefer to avoid that ATM.